src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,577 @@
+          (labeled 'lon_deg' or 'lon_rad'), and altitude (labeled 'alt_ft' or 'alt_m'). An 
+          additional column for ETAs may be included (labeled 'time_unix). Note that while
+          'flight_file' is optional, either 'flight_file' or 'flight_plan' must be specified.
+        flight_plan : Optional, dict[str, numpy array]
+          Dictionary to specify waypoint information. Necessary keys must include the following
+          with units specified: latitude ('lat_deg' or 'lat_rad'), longitude 
+          (labeled 'lon_deg' or 'lon_rad'), and altitude (labeled 'alt_ft' or 'alt_m'). An 
+          additional key for ETAs may be included (labeled 'time_unix). Each key must correspond
+          to a numpy array of values. Note that while 'flight_plan' is optional, either 
+          'flight_file' or 'flight_plan' must be specified.
+        flight_name : Optional, str
+          Optional string to identify flight plan.
+        aircraft_name : Optional, str
+          Optional string to identify aircraft. 
+        dt : Optional, float
+          Time step in seconds for trajectory generation
+        gravity : Optional, float
+          m/s^2, gravity magnitude
+        cruise_speed : float
+          m/s, avg speed in-between way-points
+        ascent_speed : float
+          m/s, vertical speed (up)
+        descent_speed : float
+          m/s, vertical speed (down)
+        landing_speed : float
+          m/s, landing speed when altitude < 10m
+        hovering_time : Optional, float
+          s, time to hover between waypoints
+        takeoff_time : Optional, float
+          s, additional takeoff time 
+        landing_time: Optional, float
+          s, additional landing time 
+        waypoint_weights: Optional, float
+          weights of the waypoints in nurbs calculation 
+        adjust_eta: Optional, dict 
+          Dictionary with keys ['hours', 'seconds'], to adjust route time
+        nurbs_basis_length: Optional, float
+          Length of the basis function in the nurbs algorithm
+        nurbs_order: Optional, int
+          Order of the nurbs curve
+        final_time_buffer_sec: Optional, float
+          s, defines an acceptable time range to reach the final waypoint
+        final_space_buffer_m: Optional, float
+          m, defines an acceptable distance range to reach final waypoint 
+        vehicle_model: Optional, str
+          String to specify vehicle type. 'tarot18' and 'djis1000' are supported
+        vehicle_payload: Optional, float
+          kg, payload mass
+
+    References 
+    ----------
+        References 
+    -------------
+     .. [0] M. Corbetta et al., "Real-time UAV trajectory prediction for safely monitoring in low-altitude airspace," AIAA Aviation 2019 Forum,  2019. https://arc.aiaa.org/doi/pdf/10.2514/6.2019-3514
+    """
+    events = ['TrajectoryComplete']
+    inputs = ['T','mx','my','mz']
```

Comment:
```suggestion
    inputs = []
```

Since 'T','mx','my','mz' are calculated by the controller, which is internal to the model, I think these should not be listed as inputs. Do I understand that correctly?

Reply:
I think this is an issue with our current implementation, where the "inputs" are defined in the internal "future_loading" function. They still act as inputs in the normal sense of u in the overall ProgPy framework, even though they aren't user defined. If I change the code as suggested here, I get an error, since u is still used throughout (for example, in the dx calculation on line 309). The more we review, the more I think it's time to switch to a more modular version ...